### PR TITLE
Begin migration away from params

### PR DESCRIPTION
**Pull request by xaque208**

### CODE REVIEW

**Comment by zleswomp** [comment]
Passing tests are cool.  Its a start.

**Comment by xaque208** [comment]
The undef value defaults here need to be addressed before this is ready for commit.

**Comment by igalic** [comment]
***nice***

n.b.: in @voxpupuli we have [tea](https://github.com/voxpupuli/puppet-tea) some composite types such as Port  and Unprivileged Port and UnixPath and so on

**Comment by zleswomp** [comment]
Oh sweet!  Thanks for the pointer @igalic.  I'll drag some of that in for path validation etc.  It will probably still take a few more passes over this to get it in shape for merge.

**Comment by igalic** [comment]
@zleswomp please note: prs to extend tea are very warmly welcomed!

**Comment by xaque208** [comment]
Dear me, come back to this.

**Comment by xaque208** [comment]
I think some of the data types validation will have to wait here.  What I'd like to do is replace the lib code here for IP validation with validation from native Pupept 4 types.  The IP address can come from thrnioi/ip, and the port can come from voxpupuli/tea, but combining those two looks like more work than I'd like to put in right now.  I've gotten this code running stable in my environment, so I think some sanity checking here would be good.  I'll come back to pass over this PR later.

If folks from the community would like to review this code before merge, I think this would be a good time for that.  I'll plan to squash up my commits before merge.

**Comment by xaque208** [comment]
Updating the supported OS means that the fact sets for FreeBSD are actually being loaded now, which has uncovered a few bugs that need addressing.  I'll work through those next.

**Comment by xaque208** [comment]
I think this is about as ready as its going to get for now.  Review would be appreciated.  I'll hold off merging a day or so.

File: metadata.json [hunk 1]
```diff
@@ -73,14 +73,14 @@
     {
       "operatingsystem": "FreeBSD",
       "operatingsystemrelease": [
-        "10.1",
-        "10.2"
+        "10"
```
Comment (igalic) [inline metadata.json]: 11 is out… does it still work there?

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -1,129 +1,145 @@
...
-    ].each do |operatingsystem|
-      describe "puppet class without any parameters on #{operatingsystem}" do
-        let(:facts) {
-          {
-            :operatingsystem => operatingsystem,
-            :concat_basedir => '/dne'
-          }
-        }
-        it { should contain_class('unbound') }
+  on_supported_os.each do |os, facts|
```
Comment (igalic) [inline spec/classes/init_spec.rb]: :+1:

File: metadata.json (hunk 1 above)
Comment (xaque208) [inline metadata.json]: It does, thanks.  Added.

**Pull request merged**